app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -70,6 +75,28 @@ open class HomeActivity : AppCompatActivity() {
         )
     }
 
+    fun getShareCallback(): TabsSharedCallback {
+        return object : TabsSharedCallback {
+            override fun describeContents(): Int {
```

Comment:
The callback as an arg doesn't work with the Navigation library without being Parcelable but I wasn't sure what do do here... It seems to work without anything...?

Reply:
Could we implement the callback on the HomeActivity itself? We can then just do one call of `(activity as HomeActivity).onTabsShared(tabs.size)` in the `ShareFragment.sendSendTab` call. 

That way we don't have to keep passing the callback around through all the fragments.

Reply:
Yeah that works too. I didn't know if it was "less clean" to make HomeActivity implement the interface and do it that way but it's certainly less code so 🤷‍♀

Reply:
I completely agree, this isn't the ideal fix we would like, but it helps with a few problems:
- Concentrating the fix to one area of the codebase.
- Making clean-up/removal easier (we decide to go about this)
- Fewer things for a new person to read and know what's going on.
- Potential for follow-up work based on this. (e.g. if we want different toast messages in the future, we can change that callback to be `onToastMessage(String)` as a generic version instead of it being specific to tab sharing.

Ideally, getting rid of the share fragment owning all this logic would be nice as well, but that's outside of the scope of your fix i.e. a share API that isn't dependant on the fragment you enter from OR a builder to create a share request/intent. 🤷‍♂ 

These are just thoughts so a lot of it debatable on what kind of balance we want. Happy to talk about it more! 😃 

</ramble>